File: TaskyJ.Globals.Data/DataObjects/DBUserJ.cs
using System.Runtime.Serialization;

namespace TaskyJ.Globals.Data.DataObjects
{
    public class DBUserJ : BaseEntity
    {
        [DataMember]
        public string UserName { get; set; }
        [DataMember]
        public string Password { get; set; }

        public override string ToString()
        {
            return UserName;
        }

        public override bool Equals(BaseEntity source)
        {
            if (source is DBUserJ sourcej)
            {
                return UserName == sourcej.UserName && base.Equals(source);
            }
            else
            {
                return base.Equals(source);
            }
        }

        public DBUserJ() : base() { }

        public DBUserJ(DBUserJ sourcej)
        {
            UserName = sourcej.UserName;
            Password = sourcej.Password;
        }

        public override void CopyFrom(BaseEntity source)
        {
            if (source != null)
            {
                base.CopyFrom(source);
                if (source is DBUserJ sourcej)
                {
                    UserName = sourcej.UserName;
                    Password = sourcej.Password;
                }
            }
        }
    }
}

File: TaskyJ.Interface.Angular/ClientApp/src/app/models/DBUserJ.ts
import { BaseEntity } from './BaseEntity';

export class DBUserJ implements BaseEntity {
  id: number;
  userName: string = null;
  password: string = null;

  public Equals(source: DBUserJ): boolean {
    return this.id == source.id &&
      this.userName == source.userName &&
      this.password == source.password;
  }

  public CopyFrom(source: DBUserJ): void {
    this.id = source.id;
    this.userName = source.userName;
    this.password = source.password;
  }
}
